### PR TITLE
Feat/#298 routes and progress bar

### PR DESCRIPTION
**Pull request by mlauren**
## Description
- Updates router context to update current route when page or chapter component mounts
- Adds VA progress bar component

## Original issue(s)
department-of-veterans-affairs/va-forms-system-core#298

## Screenshots
Screen Shot 2022-05-26 at 12 06 54 PM

## Definition of done
- [ x] Documentation has been updated, if applicable
- [ x] A link to the original github issue has been provided
- [ x] No sensitive information (i.e. PII/credentials/internal URLs etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/routing/Page.tsx [hunk 1]
```diff
@@ -12,11 +12,20 @@ import { RouterContext } from './RouterContext';
...
-  const listOfRoutes = useContext(RouterContext).listOfRoutes;
+  const { listOfRoutes, updateRoute } = useContext(RouterContext);
   const currentLocation = useLocation();
 
+  // const history = useHistory();
+  // updateRoute(currentLocation.pathname);
+
   const navigate = useNavigate();
 
+  useLayoutEffect(() =>
```
Comment (bjmccotter7192) [inline src/routing/Page.tsx:23]: is there a reason to use `useLayoutEffect` instead of `useEffect` here?

File: src/routing/Page.tsx [hunk 2]
```diff
@@ -12,11 +12,20 @@ import { RouterContext } from './RouterContext';
 export default function Page(props: PageProps): JSX.Element {
   const { values, submitForm } = useFormikContext();
   const formValues = values as IFormData;
-  const listOfRoutes = useContext(RouterContext).listOfRoutes;
+  const { listOfRoutes, updateRoute } = useContext(RouterContext);
   const currentLocation = useLocation();
 
+  // const history = useHistory();
```
Comment (bjmccotter7192) [inline src/routing/Page.tsx:18]: do you want to keep this here for future use or can we remove it?

File: src/routing/Router.tsx [hunk 3]
```diff
@@ -19,15 +21,19 @@ import { RouterContext, RouterContextProvider } from './RouterContext';
...
 
   return (
     <div className="row">
       <div className="usa-width-two-thirds medium-8 columns">
-        <RouterContextProvider routes={props.children}>
-          <BrowserRouter basename={props.basename}>
-            {props?.title && (
-              <FormTitle title={props.title} subTitle={props?.subtitle} />
-            )}
+        <BrowserRouter basename={props.basename}>
```
Comment (bjmccotter7192) [inline src/routing/Router.tsx:29]: *Not a code change* but was there a reason to move the context under the router?

**Review by bjmccotter7192 — COMMENTED** [review]
Only had a few comments and one general question but otherwise looking good and thanks for all the hard work on this!

File: src/routing/Router.tsx (hunk 3 above)
Comment (mlauren) [reply src/routing/Router.tsx:29]: no, it was just a stylistic choice. It doesn't really change anything either way but might need to be reordered if we go back and use history to listen for router changes.

File: src/routing/Page.tsx (hunk 1 above)
Comment (mlauren) [reply src/routing/Page.tsx:23]: It's really strange, but I see this error but only when I'm in debug mode: 
Screen Shot 2022-05-26 at 3 32 35 PM 
The error does not show up when using useLayoutEffect. I think I had switched one of them back and was experimenting with the rendering of both, so the fact that chapter and page have different uses was a mistake. I may switch back to useEffect and just warn the team that if they're in debug mode, they may see this error, but it doesn't seem to really effect anything. I think it's a bug in react.

File: src/routing/Page.tsx (hunk 2 above)
Comment (mlauren) [reply src/routing/Page.tsx:18]: I created a ticket to capture this [here](https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/349)

**Review by bjmccotter7192 — APPROVED** [review]
LGTM!

**Pull request merged**